firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/UserMetadata.java

```diff
@@ -135,6 +138,40 @@ public void setUserId(String identifier) {
     crashlyticsWorkers.diskWrite.submit(this::serializeUserDataIfNeeded);
   }
 
+  /**
+   * Returns a {@link Map<String, String>} containing all the custom keys to attach to the event.
+   * It overrides the values of app level custom keys with the values of event level custom keys if
+   * they're identical, and event keys or values that exceed 1024 characters are truncated.
+   * Combined with app level custom keys, the map is restricted to 64 key value pairs.
+   *
+   * @param eventKeys a {@link Map<String, String>} representing event specific keys.
+   * @return a {@link Map<String, String>} containing all the custom keys to attach to the event.
+   */
+  public Map<String, String> getCustomKeys(Map<String, String> eventKeys) {
+    // In case of empty event keys, preserve existing behavior.
+    if (eventKeys.isEmpty()) return customKeys.getKeys();
```

Comment:
Nit: Google style always uses `{ }` even for 1 liners like this go/java-style#s4.1.1-braces-always-used